packages/lib/src/load_plugin_list.ts

```diff
@@ -0,0 +1,134 @@
+import fs from "fs-extra";
+import path from "path";
+import { logger } from "./logging";
+import * as libFileOps from "./file_ops";
+
+/**
+ * Searches for local plugins in the plugins/ and external_plugins/ directories
+ * and adds them to the plugin list if they contain a package.json file.
+ *
+ * @param pluginList - Map of plugin names to their file paths
+ * @param pluginListPath - Path to the JSON file storing the plugin list
+ * @returns Promise that resolves when plugin discovery is complete
+ */
+async function findLocalPlugins(pluginList: Map<string, string>, pluginListPath: string) {
+	// Check folders for plugins
+	const pluginFolders = [
+		"plugins",
+		"external_plugins",
+	];
+
+	let has_changed = 0;
+	for (const pluginFolder of pluginFolders) {
+		let pluginNames;
+		try {
+			pluginNames = await fs.readdir(pluginFolder);
+		} catch (err: any) {
+			if (err.code === "ENOENT") {
+				// Skip if folder doesn't exist
+				continue;
+			}
+			throw err;
+		}
+		for (const pluginName of pluginNames) {
+			if (pluginList.has(pluginName)) {
+				continue;
+			}
```

Comment:
Not sure. I don't think its needed.

Reply:
The value I see is in the following two cases:
1) A local plugin is moved after already being added to the plugin list; this will cause the new path to not be added, and then the old path is removed with a warning emited for the non-existing path. It will be unclear to the user why it says the plugin was removed for not existing and emiting a warning here would help.
2) A local plugin exists at the same time as an npm installed plugin; the user may forget about the local copy and run npm update to get the latest copy, the user may become confussed as to why the plugin did not update on a reload because the npm version was silently ignored.